fetcher/src/job/job.processor.ts

```diff
@@ -18,17 +22,42 @@ export class JobProcessor extends WorkerHost {
     if (keys.length == 0 || keys.length > 1) {
       throw new FetcherError(FetcherErrorCode.UnexpectedNumberOfJobs, String(keys.length))
     } else {
-      const aggregatorHash = keys[0]
-      const aggregatorId = inData[aggregatorHash].aggregatorId
-      const feeds = inData[aggregatorHash].feeds
+      const adapterHash = keys[0]
+      const aggregatorId = inData[adapterHash].aggregatorId
+      const feeds = inData[adapterHash].feeds
       const data = await fetchData(feeds, this.logger)
       const aggregate = aggregateData(data)
+      const threshold = inData[adapterHash].threshold
+      const absoluteThreshold = inData[adapterHash].absoluteThreshold
+
+      const oracleAddress = inData[adapterHash].address
+      const aggregatorHash = inData[adapterHash].aggregatorHash
 
       try {
+        const { value: lastSubmission } = await fetchDataFeed({
+          aggregatorHash,
+          logger: this.logger
+        })
         let response = await insertMultipleData({ aggregatorId, timestamp, data })
         this.logger.debug(response)
-        response = await insertAggregateData({ aggregatorId, timestamp, value: aggregate })
+        response = await insertAggregateData({
+          aggregatorId,
+          timestamp,
+          value: aggregate
+        })
         this.logger.debug(response)
+        const outData: IDeviationData = {
+          timestamp: timestamp,
+          submission: aggregate,
+          oracleAddress
+        }
+        if (shouldReport(Number(lastSubmission), aggregate, 8, threshold, absoluteThreshold)) {
```

Comment:
Most likely the decimal value should be 8, but the actual settings that defines it is inside of adapter file. You should be able to get it the same way as threshold for example.